File: problems/0020-valid-parentheses/python/solution.py
def isValid(s):
    """Valid Parentheses — https://leetcode.com/problems/valid-parentheses/"""
    pairs = {")": "(", "]": "[", "}": "{"}
    stack = []
    for c in s:
        if c in pairs:
            if not stack or stack.pop() != pairs[c]:
                return False
        else:
            stack.append(c)
    return not stack

File: problems/0020-valid-parentheses/go/solution.go
package solution

// Valid Parentheses — https://leetcode.com/problems/valid-parentheses/
func isValid(s string) bool {
	pairs := map[byte]byte{')': '(', ']': '[', '}': '{'}
	stack := []byte{}
	for i := 0; i < len(s); i++ {
		c := s[i]
		if open, ok := pairs[c]; ok {
			if len(stack) == 0 || stack[len(stack)-1] != open {
				return false
			}
			stack = stack[:len(stack)-1]
		} else {
			stack = append(stack, c)
		}
	}
	return len(stack) == 0
}
